utils/python/CIME/build.py

```diff
@@ -301,6 +305,24 @@ def case_build(caseroot, case=None, testmode=False, sharedlib_only=False, model_
 
     if not model_only:
         logs = build_libraries(exeroot, caseroot, cimeroot, libroot, mpilib, lid, machines_file)
+        if sharedlib_only and comp_lnd == "clm" and not "clm4_0" in clm_config_opts:
```

Comment:
clm4_5 and clm5_0 can be built as shared libraries for testing.  However clm4_0, dlnd, and slnd cannot.  We may be able to move this into build_libraries but I would like to keep it here for now and make a clean up pass after we have a little more experience building with cime external components.